Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -887,20 +790,54 @@ protected void FilterThemesByDimensions()
         /// </summary>
         /// <param name="action"></param>
         /// <returns></returns>
-        protected virtual bool ShouldListen(MixedRealityPointerEventData eventData, out bool isGrab)
+        protected virtual bool ShouldListen(InputEventData data)
         {
-            if (eventData.Pointer as IMixedRealityNearPointer != null)
+            if (!(HasFocus || IsGlobal))
             {
-                isGrab = true;
-                return true;
+                return false;
             }
-            else
+
+            // Special case: Make sure that we are not being focused only by a PokePointer, since PokePointer
+            // dispatches touch events and should not be dispatching button presses like select, grip, menu, etc.
+            int pointerCount = 0;
+            int pokePointerCount = 0;
+            for (int i = 0; i < pointers.Count; i++)
             {
-                isGrab = false;
+                if(pointers[i].InputSourceParent.SourceId == data.SourceId)
+                {
+                    pointerCount++;
+                    if (pointers[i] is PokePointer)
+                    {
+                        pokePointerCount++;
+                    }
+                }
             }
 
-            bool isListening = HasFocus || IsGlobal;
-            return eventData.MixedRealityInputAction == InputAction && isListening;
+            if (pointerCount > 0 && pointerCount == pokePointerCount)
+            {
+                return false;
+            }
+
+            return data.MixedRealityInputAction == InputAction;
+        }
+
+        /// <summary>
+        /// Returns true if the inputeventdata is being dispatched from a near pointer
+        /// </summary>
+        /// <param name="eventData"></param>
+        /// <returns></returns>
+        private bool IsInputFromNearInteraction(InputEventData eventData)
+        {
+            bool isAnyNearpointerFocusing = false;
+            for (int i = 0; i < pointers.Count; i++)
+            {
+                if (pointers[i] is IMixedRealityNearPointer && pointers[i].InputSourceParent.SourceId == eventData.InputSource.SourceId)
```

Comment:
I would turn this if around so you do the source id comparison first and if that one returns false it doesn't have to do the cast

Reply:
Good point. I'll do this above as well in ShouldListen.

Reply:
Done.